### PR TITLE
[JBPM-8784] Avoid indexing dot files

### PR DESCRIPTION
**Pull request by marcuscarvalho**
This issue is addressed by the following JIRA: 

https://issues.jboss.org/browse/JBPM-8784

### CODE REVIEW

**Comment by kie-ci5** [comment]
Can one of the admins verify this PR? Comment with 'ok to test' to start the build.

**Comment by domhanak** [comment]
ok to test
@marcuscarvalho Thank you for the PR, could you please include a test that could verify these changes?

File: uberfire-extensions/uberfire-metadata/uberfire-metadata-commons-io/src/main/java/org/uberfire/ext/metadata/io/IOServiceIndexedImpl.java [hunk 1]
```diff
@@ -430,8 +430,11 @@ private void queueRenameEvent(final WatchContext context,
                                                       final IndexerDispatcher dispatcher) throws DisposedException {
                             final Path sourcePath = context.getOldPath();
                             final Path destinationPath = context.getPath();
-                            dispatcher.offer(new RenamedFileEvent(sourcePath,
-                                                                  destinationPath));
+
```
Comment (domhanak) [inline uberfire-extensions/uberfire-metadata/uberfire-metadata-commons-io/src/main/java/org/uberfire/ext/metadata/io/IOServiceIndexedImpl.java]: This method  `private void queueRenameEvent(final WatchContext context, final IndexerDispatcher dispatcher)` is not covered by tests.

File: uberfire-extensions/uberfire-metadata/uberfire-metadata-commons-io/src/main/java/org/uberfire/ext/metadata/io/IOServiceIndexedImpl.java [hunk 2]
```diff
@@ -421,7 +421,7 @@ private void queueDeleteEvent(WatchEvent object,
                                                       final IndexerDispatcher dispatcher) throws DisposedException {
                             final Path oldPath = context.getOldPath();
                             // ignore delete events for dot files, because dot files are not indexed
-                            if (!oldPath.getFileName().toString().startsWith(".")) {
+                            if (!isDotFile(oldPath)) {
```
Comment (domhanak) [inline uberfire-extensions/uberfire-metadata/uberfire-metadata-commons-io/src/main/java/org/uberfire/ext/metadata/io/IOServiceIndexedImpl.java]: `dispatcher.offer(new DeletedFileEvent(oldPath))` int this if-branch is not covered in test

File: uberfire-extensions/uberfire-metadata/uberfire-metadata-commons-io/src/main/java/org/uberfire/ext/metadata/io/IOServiceIndexedImpl.java [hunk 3]
```diff
@@ -464,7 +467,7 @@ private void queueCreationAndModificationEvent(final Set<Path> eventRealPaths,
                                 final WatchContext context = ((WatchContext) event.context());
                                 if (event.kind() == ENTRY_MODIFY || event.kind() == ENTRY_CREATE) {
                                     final Path path = context.getPath();
-                                    if (!path.getFileName().toString().startsWith(".")) {
+                                    if (!isDotFile(path)) {
                                         eventRealPaths.add(path);
```
Comment (domhanak) [inline uberfire-extensions/uberfire-metadata/uberfire-metadata-commons-io/src/main/java/org/uberfire/ext/metadata/io/IOServiceIndexedImpl.java]: `eventRealPaths.add(path);` is not covered by test

File: uberfire-extensions/uberfire-metadata/uberfire-metadata-commons-io/src/main/java/org/uberfire/ext/metadata/io/IOServiceIndexedImpl.java [hunk 4]
```diff
@@ -510,6 +513,11 @@ void deleteRepositoryFiles(Path path,
                      options);
     }
 
+    private boolean isDotFile(Path path) {
```
Comment (adrielparedes) [inline uberfire-extensions/uberfire-metadata/uberfire-metadata-commons-io/src/main/java/org/uberfire/ext/metadata/io/IOServiceIndexedImpl.java]: This is up to you, but I think I would change this for `isIgnored`. Great job @marcuscarvalho

File: uberfire-extensions/uberfire-metadata/uberfire-metadata-commons-io/src/main/java/org/uberfire/ext/metadata/io/IOServiceIndexedImpl.java (hunk 1 above)
Comment (marcuscarvalho) [reply uberfire-extensions/uberfire-metadata/uberfire-metadata-commons-io/src/main/java/org/uberfire/ext/metadata/io/IOServiceIndexedImpl.java]: Hi, @domhanak thanks for your review. I have added some unit tests. I had to extract some code to be able to test it.

File: uberfire-extensions/uberfire-metadata/uberfire-metadata-commons-io/src/main/java/org/uberfire/ext/metadata/io/IOServiceIndexedImpl.java (hunk 2 above)
Comment (marcuscarvalho) [reply uberfire-extensions/uberfire-metadata/uberfire-metadata-commons-io/src/main/java/org/uberfire/ext/metadata/io/IOServiceIndexedImpl.java]: I have added a unit test for it as well

File: uberfire-extensions/uberfire-metadata/uberfire-metadata-commons-io/src/main/java/org/uberfire/ext/metadata/io/IOServiceIndexedImpl.java (hunk 3 above)
Comment (marcuscarvalho) [reply uberfire-extensions/uberfire-metadata/uberfire-metadata-commons-io/src/main/java/org/uberfire/ext/metadata/io/IOServiceIndexedImpl.java]: For nullable paths, I am returning true from isIgnored method.

File: uberfire-extensions/uberfire-metadata/uberfire-metadata-commons-io/src/main/java/org/uberfire/ext/metadata/io/IOServiceIndexedImpl.java (hunk 4 above)
Comment (marcuscarvalho) [reply uberfire-extensions/uberfire-metadata/uberfire-metadata-commons-io/src/main/java/org/uberfire/ext/metadata/io/IOServiceIndexedImpl.java]: I agree, "isIgnored" is more readable. Thanks for your review @adrielparedes :)

**Comment by domhanak** [comment]
Jenkins test this please

**Comment by domhanak** [comment]
@marcuscarvalho Thank you for the changes. Running the build now, will trigger FDB once I see it pass

**Comment by domhanak** [comment]
Jenkins execute full downstream build

**Comment by marcuscarvalho** [comment]
Thanks @adrielparedes and @domhanak

**Comment by jesuino** [comment]
Jenkins execute full downstream build

**Comment by domhanak** [comment]
Jenkins execute full downstream build

@marcuscarvalho I am running it one more time, but imho the test failures should be looked into.

**Comment by caponetto** [comment]
Jenkins execute full downstream build

**Comment by caponetto** [comment]
Jenkins execute full downstream build

**Comment by caponetto** [comment]
Can we merge this one?
@paulovmr

**Pull request merged**